### ISSUE TITLE
Can't execute page scripts

### ISSUE THREAD

**Issue by pmsferreira** [jsdom/jsdom#2432]
### Basic info:

- **Node.js version:** 9.4.0
- **jsdom version:** 13.0.0

### Minimal reproduction case

const jsdom = require("jsdom");
const { JSDOM } = jsdom;

var options = {
     resources: "usable",
     runScripts: "dangerously"
 };

JSDOM.fromURL("https://www.abola.pt/", options).then(result => {
  console.log('\n', result.window)
});

This page has 24 scripts (obtained with result.window.document.scripts.length) but I always get errors on console such as:

- [TypeError: Cannot read property 'fetch' of null]

- [ReferenceError: MessageChannel is not defined]

Could you give me some help or advice?

Best regards,
Pedro Ferreira

**Comment by domenic** [comment]
Unfortunately we don't support fetch() or MessageChannel yet, so this is to be expected. Sorry that jsdom is not yet featureful enough for your use cases.

**Comment by pmsferreira** [comment]
@domenic thanks for response. Could you give me a website url example that works with jsdom?

**Comment by molant** [comment]
https://example.com


________________________________

**Comment by pmsferreira** [comment]
@molant yes it works but it seems that that website don't have scripts to execute...

**Comment by molant** [comment]
@pmsferreira https://webhint.io uses jsdom for the online scanner with pretty good results. There are a few websites that fail for features that are not implemented, but we are quite happy with it. Just open the history of your browser and see what works.
